src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -423,7 +423,18 @@ void GenericConfigurationManagerImpl<ImplClass>::NotifyOfAdvertisementStart()
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetRegulatoryLocation(uint8_t & location)
 {
-    return GetLocationCapability(location);
+    uint32_t value;
+    if (CHIP_NO_ERROR != ReadConfigValue(ConfigClass::kConfigKey_RegulatoryLocation, value))
+    {
+        ReturnErrorOnFailure(GetLocationCapability(location));
+        StoreRegulatoryLocation(location);
+    }
+    else
+    {
+        location = static_cast<uint8_t>(value);
```

Comment:
how do we know that value fits in a uint8_t? Could we add a check and return an error if return value is outside of the range?

Reply:
`CanCastTo<uint8_t>(value)`